enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/EagerizationAcceptanceTest.scala

```diff
@@ -2714,6 +2714,41 @@ class EagerizationAcceptanceTest
     assertNumberOfEagerness(query, 1)
   }
 
+  test("merge and pattern comprehension readwrite conflict requires eager") {
+    val query =
+      """
+        |CREATE (node)
+        |WITH node
+        |UNWIND [{ value: 'apples' }, { value: 'oranges' }] as tags
+        |MERGE (tag { value: tags.value})
+        |MERGE (node)-[:HAS_TAG {relProp1: 'relProp1', relProp2: 'relProp2'}]->(tag)
+        |WITH DISTINCT node
+        |RETURN [(node)-[:HAS_TAG {relProp1: 'relProp1', relProp2: 'relProp2'}]->(t) | t.value] as tags
+      """.stripMargin
+
+    val result = executeWithCostPlannerAndInterpretedRuntimeOnly(query)
+    result.toList should equal(List(Map("tags" -> List("apples", "oranges"))))
+    assertNumberOfEagerness(query, 1)
+  }
+
+  test("merge and optional match readwrite conflict requires eager") {
+    val query =
+      """
+        |CREATE (node:Node {id: 'xyzzy'})
+        |WITH node
+        |UNWIND [{ value: 'apples' }, { value: 'oranges' }] as tags
+        |MERGE (tag:Tag { value: tags.value})
+        |MERGE (node)-[:HAS_TAG]->(tag)
+        |WITH DISTINCT node
+        |OPTIONAL MATCH (node)-[:HAS_TAG]->(t:Tag)
+        |RETURN COLLECT(t.value) as tags
+      """.stripMargin
+
+    val result = executeWithCostPlannerAndInterpretedRuntimeOnly(query)
+    result.toList should equal(List(Map("tags" -> Vector("apples", "oranges"))))
+    assertNumberOfEagerness(query, 2)
```

Comment:
I am wondering why this query has an eager between the two MERGEs and the one above has not. I just checked and this was not introduced with your changes. It had an eager there before while the other one didn't. Any idea as to why this is the case? Is one query missing an eager or is this one producing an eager to much?

Reply:
Interesting, no idea why that would differ, will try to look into it.

Reply:
Have created a card about it